libpod/runtime.go

```diff
@@ -164,14 +164,22 @@ var (
 		CgroupManager: CgroupfsCgroupsManager,
 		HooksDir:      hooks.DefaultDir,
 		StaticDir:     filepath.Join(storage.DefaultStoreOptions.GraphRoot, "libpod"),
-		TmpDir:        "/var/run/libpod",
+		TmpDir:        getDefaultTmpDir(),
 		MaxLogSize:    -1,
 		NoPivotRoot:   false,
 		CNIConfigDir:  "/etc/cni/net.d/",
 		CNIPluginDir:  []string{"/usr/libexec/cni", "/usr/lib/cni", "/opt/cni/bin"},
 	}
 )
 
+func getDefaultTmpDir() string {
+	if os.Getuid() == 0 {
+		return "/var/run/libpod"
+	}
+
+	return filepath.Join(os.Getenv("XDG_RUNTIME_DIR"), "libpod", "tmp")
```

Comment:
I think it's worth checking that `XDG_RUNTIME_DIR` is defined before we return this - we don't want to stick things in `/libpod/tmp` if it's ""

Reply:
Right, we should default to /run/user/UID if it is not set.

Reply:
> Right, we should default to /run/user/UID if it is not set.

Is that covered by some cross-distro spec?  I don't see it in [the FHS][1].  And at least on Gentoo, unprivileged users can't create it on their own:

```console
$ mkdir -p /run/user/$(id -u)
mkdir: cannot create directory ‘/run/user’: Permission denied
```

Using `$TMP` or something from XDG may be more portable.

[1]: http://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html

Reply:
I've changed the fallback to look for `/run/user/UID`.  If it exists then we use it, otherwise we use `/tmp/user/UID`